src/fmu/dataio/providers/_objectdata_base.py

```diff
@@ -0,0 +1,393 @@
+        objres = self._derive_objectdata()
+        if self.dataio.forcefolder and not self.dataio.forcefolder.startswith("/"):
+            msg = (
+                f"The standard folder name is overrided from {objres.efolder} to "
+                f"{self.dataio.forcefolder}"
+            )
+            objres.efolder = self.dataio.forcefolder
+            logger.info(msg)
+            warn(msg, UserWarning)
+
+        meta = self.metadata  # shortform
+
+        meta["name"] = namedstratigraphy.name
+        meta["stratigraphic"] = namedstratigraphy.stratigraphic
+        meta["offset"] = namedstratigraphy.offset
+        meta["alias"] = namedstratigraphy.alias
+        meta["top"] = namedstratigraphy.top
+        meta["base"] = namedstratigraphy.base
+
+        content, content_spesific = self._process_content()
+        meta["content"] = content
+        if content_spesific:
+            meta[self.dataio._usecontent] = content_spesific
+
+        meta["tagname"] = self.dataio.tagname
+        meta["format"] = objres.fmt
+        meta["layout"] = objres.layout
+        meta["unit"] = self.dataio.unit
+        meta["vertical_domain"] = list(self.dataio.vertical_domain.keys())[0]
+        meta["depth_reference"] = list(self.dataio.vertical_domain.values())[0]
+        meta["spec"] = objres.spec
+        meta["bbox"] = objres.bbox
+        meta["table_index"] = objres.table_index
+        meta["undef_is_zero"] = self.dataio.undef_is_zero
+
+        # timedata:
+        dt = self._derive_timedata()
+        if isinstance(dt, TimedataLegacyFormat) and dt.time:
+            meta["time"] = [asdict(v) for v in dt.time]
+        elif isinstance(dt, TimedataFormat):
+            if dt.t0 or dt.t1:
+                meta["time"] = {}
+            if t0 := dt.t0:
+                meta["time"]["t0"] = asdict(t0)
+            if t1 := dt.t1:
+                meta["time"]["t1"] = asdict(t1)
+
+        meta["is_prediction"] = self.dataio.is_prediction
+        meta["is_observation"] = self.dataio.is_observation
+        meta["description"] = generate_description(self.dataio.description)
+
+        # the next is to give addition state variables identical values, and for
+        # consistency these are derived after all eventual validation and directly from
+        # the self.metadata fields:
+
+        self.name = meta["name"]
+
+        # then there are a few settings that are not in the ``data`` metadata, but
+        # needed as data/variables in other classes:
+
+        self.efolder = objres.efolder
+        self.classname = objres.classname
+        self.extension = objres.extension
+        self.fmt = objres.fmt
+        logger.info("Derive all metadata for data object... DONE")
+
+    @abstractmethod
+    def _derive_spec_and_bbox(self) -> SpecificationAndBoundingBox:
+        raise NotImplementedError
+
+    @abstractmethod
+    def _derive_objectdata(self) -> DerivedObjectDescriptor:
+        raise NotImplementedError
```

Comment:
These copy the implied style of the old single class and `derive_metadata` will call them. I expect to change these in a following PR, a bit -- at least `_derive_spec_and_bbox`, since they can and should be split into two separate methods.